include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -1617,21 +1582,23 @@ class operational_domain_impl
  * @param stats Operational domain computation statistics.
  * @return The operational domain of the layout.
  */
-template <typename Lyt, typename TT>
-operational_domain<parameter_point, operational_status>
-operational_domain_grid_search(const Lyt& lyt, const std::vector<TT>& spec,
-                               const operational_domain_params& params = {}, operational_domain_stats* stats = nullptr)
+template <typename OpDomain = operational_domain, typename Lyt, typename TT>
+OpDomain operational_domain_grid_search(const Lyt& lyt, const std::vector<TT>& spec,
+                                        const operational_domain_params& params = {},
+                                        operational_domain_stats*        stats  = nullptr)
 {
     static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
     static_assert(has_sidb_technology_v<Lyt>, "Lyt is not an SiDB layout");
     static_assert(kitty::is_truth_table<TT>::value, "TT is not a truth table");
+    static_assert(std::is_same_v<OpDomain, operational_domain> ||
+                      std::is_same_v<OpDomain, temperature_operational_domain>,
+                  "OpDomain must be operational_domain or temperature_operational_domain");
```

Comment:
I would have kept these functions as is and provided additional ones called `temperature_domain_grid_search`, `temperature_domain_random_sampling`, etc. That simplifies the caller interface and makes it harder for users to use them wrong.

Reply:
Sure! We can do this.